.github/workflows/conflicts.yml

```diff
@@ -1,9 +1,6 @@
 name: Add a conflict label if PR needs to rebase
 
-on:
-  push:
-  pull_request_target:
-    types: [synchronize]
+on: pull_request
```

Comment:
I don't remember, but It might require some permissions (write) where `pull_request` is read-only. Let's try and see.

Reply:
Btw, this isn't working `if: github.repository == 'frrouting/frr'`?

Reply:
It skips running the job, and it sends an email telling you that it did that. It's super annoying :)

Reply:
> I don't remember, but It might require some permissions (write) where `pull_request` is read-only. Let's try and see.

So the story is this, if a PR is from a public forked repository, then the GITHUB_TOKEN is only granted read permissions for everything. For PRs created on the project repo the default permissions are used (which are permissive).

When `pull_request_target` is used write permissions are granted to GITHUB_TOKEN, but the GITHUB_SHA/REF are from the base of the pull request and the workflows are run from there. This is to prevent someone from forking the repo and creating a PR modifying a workflow that then maliciously modifies the project repo with it's write permissions.

Additionally PRs with conflicts do not trigger workflows waiting on `pull_request` but conflicted PRs do trigger workflows waiting on `pull_request_target`. So  `pull_request_target` is the correct choice.

It's probably worth asking, since we are using a third party action here, should we trust it with write permission to our project repo? This action is just adding a label right? The PR will always be marked as having conflicts if it has conflicts. I'm not sure the value of the adding/removing a label is worth the security exposure.

Here's the warning from the github docs on this: ```
Warning: For workflows that are triggered by the pull_request_target event, the GITHUB_TOKEN is granted read/write repository permission unless the permissions key is specified and the workflow can access secrets, even when it is triggered from a fork. Although the workflow runs in the context of the base of the pull request, you should make sure that you do not check out, build, or run untrusted code from the pull request with this event. Additionally, any caches share the same scope as the base branch. To help prevent cache poisoning, you should not save the cache if there is a possibility that the cache contents were altered. For more information, see "[Keeping your GitHub Actions and workflows secure: Preventing pwn requests](https://securitylab.github.com/research/github-actions-preventing-pwn-requests)" on the GitHub Security Lab website.
```